gateway/api.go

```diff
@@ -3268,3 +3271,21 @@ func updateOASServers(spec *APISpec, conf config.Config, apiDef *apidef.APIDefin
 	newAPIURL := getAPIURL(*apiDef, conf)
 	oasObj.UpdateServers(newAPIURL, oldAPIURL)
 }
+
+// GetSessionLifetime defines the TTL of the key
+func GetSessionLifetime(sess *user.SessionState, gw *Gateway) int64 {
+	var lifetime int64
+	if sess != nil && len(sess.AccessRights) > 0 {
+		for apiID, _ := range sess.AccessRights {
+			spec := gw.getApiSpec(apiID)
+			if spec != nil {
+				sessionLifeTime := sess.Lifetime(spec.GetSessionLifetimeRespectsKeyExpiration(), spec.SessionLifetime, gw.GetConfig().ForceGlobalSessionLifetime, gw.GetConfig().GlobalSessionLifetime)
+				// uses the greater lifetime
+				if sessionLifeTime > lifetime {
+					lifetime = sessionLifeTime
+				}
+			}
+		}
+	}
+	return lifetime
```

Comment:
this should returns -1 if the lifetime value is 0

Reply:
@tbuchaillot zero is fine. The redis handler considers 0 as never expiry key

Reply:
Accordingly to the documentation: `// Zero expiration means the key has no expiration time.`